src/main/scala/common/master_adapter.scala

```diff
@@ -0,0 +1,128 @@
+package sodor.common
+
+import chisel3._
+import chisel3.util._
+
+import freechips.rocketchip.config._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.diplomaticobjectmodel.logicaltree.{LogicalTreeNode}
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.subsystem.{RocketCrossingParams}
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.util._
+import freechips.rocketchip.tile._
+import freechips.rocketchip.amba.axi4._
+
+class SodorMasterAdapter(implicit p: Parameters, val conf: SodorConfiguration) extends LazyModule {
+  // The node exposed to the crossbar
+  val node = TLIdentityNode()
+
+  // The client node (only one inflight request supported for Sodor)
+  // This node handles core requests to addresses not managed in the tile-local scratchpad
+  val masterNode = TLClientNode(Seq(TLMasterPortParameters.v1(
+    clients = Seq(TLClientParameters(
+      name = "sodor-mmio-master",
+      sourceId = IdRange(0, 1)
+    ))
+  )))
+
+  // Connect nodes
+  (node := TLBuffer() := masterNode)
+
+  lazy val module = new SodorMasterAdapterImp(this)
+}
+
+class SodorMasterAdapterImp(outer: SodorMasterAdapter) extends LazyModuleImp(outer) {
+  implicit val conf = outer.conf
+
+  val io = IO(new Bundle() {
+    val dport = Flipped(new MemPortIo(data_width = conf.xprlen))
+  })
+
+  val (tl_out, edge) = outer.masterNode.out(0)
+
+  // Register
+  // State
+  val s_ready :: s_active :: s_inflight :: Nil = Enum(3)
+  val state = RegInit(s_ready)
+  // Address and signedness of the request to be used by LoadGen
+  val a_address_reg = RegInit(0.U(io.dport.req.bits.addr.getWidth.W))
+  val a_signed_reg = RegInit(false.B)
+  // Request register - store the request when the input port fires to avoid value changes when sending the TileLink request
+  val req_address_reg = RegInit(0.U(io.dport.req.bits.addr.getWidth.W))
+  val req_size_reg = RegInit(0.U(2.W))
+  val req_data_reg = RegInit(0.U(io.dport.req.bits.data.getWidth.W))
```

Comment:
Based on the state machine, resetting these registers seems to be unnecessary.

Reply:
Done